automation/DataAggregator/schema.sql

```diff
@@ -120,6 +120,7 @@ CREATE TABLE IF NOT EXISTS http_redirects(
   frame_id INTEGER,
   response_status INTEGER NOT NULL,
   response_status_text TEXT NOT NULL,
+  headers TEXT NOT NULL,
```

Comment:
It would be great to have this in in the responses table, rather than redirects. The headers are actually a property of the response, and from your description it sounds like the column is otherwise empty when there are redirects. Is that right?

However, I suspect this isn't straightforward since the actual data is available in the redirect listener, and you'd need to save it alongside data saved in a separate response listener. If I'm missing something and you see a straightforward way to do this please do. Otherwise, there's no need to add complexity here.

Reply:
That's a good question. In general, when a redirect is triggered, we get pretty much everything as for an response, except for the content or contentHash. So I could modify the patch in a way that a redirect would trigger two writes, one in the redirect table and one in the response table. However, both tables are currently already very similar, except for the missing old and new URL and request id fields that are only present in the redirect table and we would then save the data twice.

The other option would be to merge the response table with the redirect table, but then some fields would be empty depending on whether it's an redirect or not.

In my opinion, having the headers in the redirect table is currently the cleanest solution, but other more invasive changes could be possible as well.

Reply:
I agree. This seems like the best approach to me as well. Thanks for thinking through it.